### ISSUE TITLE
Redundant device definitions for Philio

### ISSUE THREAD

**Issue by rom169** [cdjackson/HABmin#271]
The following Philio device definitions are redundant in products.xml:
<Type>0002</Type><Id>0002</Id>
<Type>0002</Type><Id>000D</Id>
I assume this is the reason, why habmin product explorer does not work correctly in Philio section.
My suggestion would be to delete the unused redundant definitions, I mean the ones the device detection algorithm won’t use.
Regards
rom169

**Comment by cdjackson** [comment]
Well spotted…

So, which ones are wrong? There’s no way to tell - I could just delete them all and see who complains :)

**Comment by rom169** [comment]
I cannot tell which ones are wrong either, as I have just a PST-02 Type 0002 ID 000E.
Isn’t it possible to check the device detection algorithm code of openhab to see which definition is selected and then delete the other?

**Comment by cdjackson** [comment]
There’s no way to know what type/id equals what model, so we’d basically need everyone to advise of their type/id and model…

I think there’s only one set of numbers duplicated - your set aren’t (I think?) so it should work fine, but there’s no way to know which of the others is wrong.

**Comment by rom169** [comment]
Yes, my device is working fine, just the problem with device explorer for Philio is a little bit annoying.
If a new device gets connected to openhab, would openhab browse the poducts.xml from top to bottom or from bottom to top? Knowing this, the definitions never used could be deleted.

**Comment by cdjackson** [comment]
I don’t know - it all gets read into a list using the xml reader, so I wouldn’t like to say which way it is - or even if it’s the same each time (which it probably is, but I’m not sure).

**Comment by rom169** [comment]
I checked the issues and pull requests related to PST. There are two referring to the following device:
http://www.pepper1.net/zwavedb/device/585
But I cannot find any product description on Philio homepage or on z-wave.com for this device.
There are just the PSM02 multisensor (zwave) and the PST02 multisensor (zwave+).
Could it be that the device documented on pepper does not exists?
If so, I can check which definition files are the correct ones for PSM02 and PST02 on z-wave.com and Philio. The other ones could be deleted then.

**Comment by rom169** [comment]
Hi cdjackson,
I probably find some time on the weekend to check available devices and configurations. But this makes just sense, if you would support the suggested approach of my last post?
Regards
rom169

**Comment by cdjackson** [comment]
Typically, we don’t use the device IDs from pepper - normally the IDs come from users. So, someone has entered this data - maybe they got the wrong device, or maybe something else is wrong, but it’s not possible to know…. Certainly pepper isn’t always correct for device ids...

**Comment by rom169** [comment]
Hi cdjackson,

I was reviewing all available documentation (z-wave.com, pepper, philio-tech homepage) and from my understanding we have answer the following questions:

Is mapping Type 0002 ID 0002 to psm02.xml or to pst02a.xml correct?
The two xml files are almost the same, there are just a view additional command classes and some different defaults in pst02a.xml. According to the original philio documentation the numbers of parameters and associations refer to a psm device. I would suggest to delete pst02a.xml file and the Type 0002 ID 0002  mapping to this file.

Is mapping Type 0002 ID 000D to pst02.xml or to pst02b.xml correct?
The two xml files are different, the number of parameters and associations and some defaults are different. According to the original philio documentation the numbers of parameters and associations in pst02.xml are correct. I would suggest to delete pst02b.xml and the Type 0002 ID 000d mapping to this file.

Reviewing the official philio documentations, I found also some minor differences of the default/minimum values in pst02.xml file:
Parameter 3 Default should be 80
Parameter 4 Default should be 99
Parameter 13 Minimum should be 0
Parameter 22 Default should be 0

Regards
rom169

**Comment by rom169** [comment]
Hi cdjackson,
are you considering the suggested changes or should I close this issue?
Regards
rom169
